nautobot/core/api/schema.py

```diff
@@ -145,6 +234,18 @@ def get_serializer_ref_name(self, serializer):
             ref_name = ref_name[: -len("Serializer")]
         return ref_name
 
+    def resolve_serializer(self, serializer, direction):
+        """
+        Re-add required `id` field on bulk_partial_update action.
+
+        drf-spectacular clears the `required` list for any partial serializers in its `_map_basic_serializer()`,
+        but Nautobot bulk partial updates require the `id` field to be specified for each object to update.
+        """
+        component = super().resolve_serializer(serializer, direction)
+        if self.is_bulk_action and self.is_partial_action and direction == "request":
+            component.schema["required"] = ["id"]
+        return component
```

Comment:
I hate that we have to do all of this and if there's any reasonable expectation that we can undo this in v2, can we please add "v2 TODO" for these?

Reply:
I think bulk-PATCH will always require `id` as a field, as how else to identify the specific instances being patched? Unless we want to get rid of these bulk operations entirely in v2.

Reply:
I suppose what I meant is if there's a way we can get away with it without requiring so much high touch. But perhaps now that this is solved it's less of a concern.